libdevice/cmake/modules/SYCLLibdevice.cmake

```diff
@@ -49,10 +49,13 @@ endif()
 if ("NVPTX" IN_LIST LLVM_TARGETS_TO_BUILD)
   string(APPEND sycl_targets_opt ",nvptx64-nvidia-cuda")
   list(APPEND compile_opts
+    "-Xsycl-target-backend=nvptx64-nvidia-cuda"
     "-fno-sycl-libspirv"
+    "-Wno-no-libspirv-hip-cuda"
     "-fno-bundle-offload-arch"
     "-nocudalib"
-    "--cuda-gpu-arch=sm_50")
+    "--cuda-gpu-arch=sm_50"
+    "-Wno-unused-command-line-argument")
```

Comment:
I do understand that this is one option to get rid of all these command line arguments.
However, I am not sure if we should not better focus on refactoring the CMake file by introducing new variables etc. to get rid of these warnings.

Reply:
In theory this is a good idea, but the problem is that we are compiling the obj files for multiple triples per compiler invocation. See something like this:
```
cd /home/hugh/llvm/build/tools/libdevice && /home/hugh/llvm/build/bin/clang-19 -fsycl -c -Wno-sycl-strict -Wno-undefined-internal -sycl-std=2020 -Xsycl-target-backend=nvptx64-nvidia-cuda -fno-sycl-libspirv -Wno-no-libspirv-hip-cuda -Xsycl-target-backend=nvptx64-nvidia-cuda -fno-bundle-offload-arch -nocudalib --cuda-gpu-arch=sm_50 -fsycl-targets=spir64_x86_64-unknown-unknown,spir64_gen-unknown-unknown,spir64_fpga-unknown-unknown,spir64-unknown-unknown,spirv64-unknown-unknown,nvptx64-nvidia-cuda /home/hugh/llvm/libdevice/fallback-bfloat16.cpp -o /home/hugh/llvm/build/./lib/libsycl-fallback-bfloat16.o
```

When using target-specific compilation flags, these flags will not be used by other triples. This means that if we want to remove the warnings about unused args, we can either
1. Change the action so we compile objects per triple and then link them. I think this is overkill personally, especially since the way this is build is likely to change soon.
2. Tell the compiler not to emit warnings for unused command line args, ie when compiling for a triple in the `-fsycl-targets` list where target specific flags do not relate to the triple. I think this is straightforward and simple and doesn't interfere with the current work to restructure the libdevice build process.

Let me know your thoughts

Reply:
@hdelan 

Ok thanks, I do see now why it makes more sense to add it and agree with 2.
The only thing that I still find a bit concerning is that we add this flag all the time to each compilation function that does not even emit .obj files.
However, we already similarly use CUDA flags when creating .bc files that are also not used.
I'm wondering therefore if it's perhaps possible to add the "-Wno-unused-command-line-argument" only to the .obj and .new-obj compilation functions and similarly remove the unused CUDA flags when compiling .bc files.
This is a heavy nit, but it helps in my opinion to keep these cmake files clean.